### ISSUE TITLE
Make compatible with CFTime

### ISSUE THREAD

**Issue by willirath** [willirath/xorca#34]
https://github.com/willirath/xorca/blob/1cab7220aa2cea8698db5d86b3e0d75449959539/xorca/lib.py#L67_L68 tries to find time dimensions by ckecking their data types. This fails if cftime objects are used on the time axis.

Road to fully support cftime:

1. We pass `use_cftime` kwarg to all the `open_dataset` calls.

1. Fix 
```python
    def _is_time_dim(ds, dim):
        return (dim in orca_names.t_dims and
                np.issubdtype(ds[dim].dtype,
                              np.datetime64))
```
to sth like
```python
    def _is_time_dim(ds, dim):
        return (dim in orca_names.t_dims and
                np.issubdtype(ds[dim].dtype,
                              'object'))
```
(But note that it would be better to actively check for valid time types. this is likely also done somewhere in xarray's plotting backend.)

**Comment by willirath** [comment]
`(... and (np.issubdtype(ds[dim].dtype, np.datetime64) or <check for cftime>))`

**Comment by janharlass** [comment]
Apparently it does work with files that have a monthly resolution in time or better - more than one time step. There is no issue with the time axis for times after somewhere 2262.

**Comment by janharlass** [comment]
But it still fails with some `'<' not supported between instances of 'NoneType' and 'NoneType'` error, when trying to read yearly data, meaning only 1 time step in a file. Could it be related to the possible chunks? Or sorting?

**Comment by willirath** [comment]
> Could it be related to the possible chunks? Or sorting?

No. It's because the type of the time dim is not recognised. (See above). XORCA makes sure that time dims are not squeezed away.

**Comment by janharlass** [comment]
XORCA checks for datetime64, which the yearly time values before year 26?? clearly are. I double checked with:  
`d1 = xr.concat( (xr.open_dataset(file.format(year=y), use_cftime=False).rename({'time_counter': 'time'}) for y in (2000, 2001, 2002, 2003)), dim="time")`   
If use_cftime=True and/or year is after 26??, they become objects and are not recognized by XORCA anymore.   
Still, if I remove the time dimension check and simply allow all time axis to be a time axis, no matter what their type might be and they will not be squeezed, it still fails.

**Comment by janharlass** [comment]
I think I found a solution. Probably improvable in coding style, but it works. All variables from the aux files are correctly transferred to the new data set and time axis are carfeuly checked. All complies to cftime. See #35.
